src/services/postService.js

```diff
@@ -119,6 +119,41 @@ async function checkLike(like, postID, userID){
     return postData;
 }
 
+async function checkTags(tags, inclusive){
+    const posts = await postDAO.scanPosts();
+    throwIfError(posts);
+    if (!tags){
+        return posts.Items;
+    }
+    tags = tags.split(',');
+    const postSet = new Set();
+    if (inclusive == 1){
+        for (const post of posts.Items){
+            for (const i of tags){
+                if (post.tags.includes(i)){
```

Comment:
This is a O(n*m*k) time complexity function which can be really bad. It depends on the numbers of posts, the tags the user gives us, and the tags assigned to the post. Hopefully the number of tags stays small.